src/amms/uniswap_v3/mod.rs

```diff
@@ -414,15 +418,17 @@ impl AutomatedMarketMaker for UniswapV3Pool {
                     current_state.tick,
                     self.tick_spacing,
                     zero_for_one,
-                )?;
+                )
+                .map_err(UniswapV3Error::from)?;
```

Comment:
All uniswap v3 math errors can be propagated with `?`. Can we apply this to all other instances?